src/control/server/instance_storage_rpc.go

```diff
@@ -238,67 +245,92 @@ func scanEngineBdevsOverDrpc(ctx context.Context, ei *EngineInstance, pbReq *ctl
 				nsd.MetaWalSize = c.HealthStats.MetaWalSize
 				nsd.RdbWalSize = c.HealthStats.RdbWalSize
 			}
+			engineRank, err := engine.GetRank()
+			if err != nil {
+				return nil, errors.Wrapf(err, "instance %d GetRank", engine.Index())
+			}
+			nsd.Rank = engineRank.Uint32()
 			c.SmdDevices = append(c.SmdDevices, nsd)
 		}
 	}
 
 	return &pbResp, nil
 }
 
-// bdevScanEngine calls either in to the private engine storage provider to scan bdevs if engine process
-// is not started, otherwise dRPC is used to retrieve details from the online engine.
-func bdevScanEngine(ctx context.Context, engine Engine, pbReq *ctlpb.ScanNvmeReq) (*ctlpb.ScanNvmeResp, error) {
-	ei, ok := engine.(*EngineInstance)
-	if !ok {
-		return nil, errors.New("not EngineInstance")
+func bdevScanEngineAssigned(ctx context.Context, engine Engine, pbReq *ctlpb.ScanNvmeReq, devList *storage.BdevDeviceList, isStarted *bool) (*ctlpb.ScanNvmeResp, error) {
+	*isStarted = engine.IsStarted()
+	if !*isStarted {
+		//ei.log.Debugf("scanning engine-%d bdev tiers while engine is down", engine.Index())
```

Comment:
Need to remove? Could make it a trace level log if it's useful sometimes.

Reply:
I will address when I have to re-push (after the parent lands).